logstash-core/src/main/java/org/logstash/common/io/MmapPageIO.java

```diff
@@ -0,0 +1,254 @@
+package org.logstash.common.io;
+
+import org.logstash.ackedqueue.Queueable;
+import org.logstash.ackedqueue.SequencedList;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+// TODO: this essentially a copy of ByteBufferPageIO and should be DRY'ed - temp impl to test file based stress test
```

Comment:
temp impl for stress tests?

if this is only used for tests, maybe it belongs in the `src/test/java/org/logstash/..` directory?